engine_test.go

```diff
@@ -201,6 +201,20 @@ var queries = []struct {
 			{int64(1), "third row"},
 		},
 	},
+	{
+		`SELECT COUNT(*), fi FROM (
```

Comment:
can you rewrite this as `SELECT fi, COUNT(*) FROM (` to see if it still works?
COUNT in orderby receives a row in Eval (which should be the buffer) and since the first item is the result of the count, it will work. But it won't if you change the order.